src/qtAliceVisionImageIO/QtAliceVisionImageIOHandler.cpp

```diff
@@ -0,0 +1,261 @@
+#include <QDebug>
+
+#include <OpenImageIO/imageio.h>
+#include <OpenImageIO/imagebuf.h>
+#include <OpenImageIO/imagebufalgo.h>
+
+#include <aliceVision/image/io.hpp>
+#include <aliceVision/image/Image.hpp>
+#include <aliceVision/image/pixelTypes.hpp>
+
+#include <iostream>
+#include <memory>
+
+
+inline const float& clamp( const float& v, const float& lo, const float& hi )
+{
+    assert( !(hi < lo) );
+    return (v < lo) ? lo : (hi < v) ? hi : v;
+}
+
+inline unsigned short floatToUShort(float v)
+{
+    return clamp(v, 0.0f, 1.0f) * 65535;
+}
+
+QtAliceVisionImageIOHandler::QtAliceVisionImageIOHandler()
+{
+    qDebug() << "[QtAliceVisionImageIO] QtAliceVisionImageIOHandler";
+}
+
+QtAliceVisionImageIOHandler::~QtAliceVisionImageIOHandler()
+{
+}
+
+bool QtAliceVisionImageIOHandler::canRead() const
+{
+    if(canRead(device()))
+    {
+        setFormat("AliceVisionImageIO");
+        return true;
+    }
+    return false;
+}
+
+bool QtAliceVisionImageIOHandler::canRead(QIODevice *device)
+{
+    QFileDevice* d = dynamic_cast<QFileDevice*>(device);
+    if(d)
+    {
+        qDebug() << "[QtAliceVisionImageIO] Can read file: " << d->fileName();
+        return true;
+    }
+    qDebug() << "[QtAliceVisionImageIO] Cannot read.";
+    return false;
+}
+
+bool QtAliceVisionImageIOHandler::read(QImage *image)
+{
+    bool convertGrayscaleToJetColorMap = true; // how to expose it as an option?
+
+    QFileDevice* d = dynamic_cast<QFileDevice*>(device());
+    if(!d)
+    {
+        qWarning() << "[QtAliceVisionImageIO] Read image failed (not a FileDevice).";
+        return false;
+    }
+    const std::string path = d->fileName().toStdString();
+
+    qDebug() << "[QtAliceVisionImageIO] Read image: " << path.c_str();
+
+    aliceVision::image::Image<aliceVision::image::RGBColor> img;
+    aliceVision::image::readImage(path, img, aliceVision::image::EImageColorSpace::LINEAR);
+
+    oiio::ImageBuf inBuf;
+    aliceVision::image::getBufferFromImage(img, inBuf);
+
+    oiio::ImageSpec inSpec = aliceVision::image::readImageSpec(path);
+    float pixelAspectRatio = inSpec.get_float_attribute("PixelAspectRatio", 1.0f);
+
+    qDebug() << "[QtAliceVisionImageIO] width:" << inSpec.width 
+            << ", height:" << inSpec.height 
+            << ", nchannels:" << inSpec.nchannels  
+            << ", pixelAspectRatio:" << pixelAspectRatio;
+    
+    qDebug() << "[QtAliceVisionImageIO] create output QImage";
+    QImage result(inSpec.width, inSpec.height, QImage::Format_RGB32);
+
+    const int nchannels = 4;
+    const oiio::TypeDesc typeDesc = oiio::TypeDesc::UINT8;
+    oiio::ImageSpec requestedSpec(inSpec.width, inSpec.height, nchannels, typeDesc);
+    oiio::ImageBuf tmpBuf(requestedSpec);
+    oiio::ROI exportROI = inBuf.roi();
+    exportROI.chbegin = 0;
+    exportROI.chend = nchannels;
+
+    // if the input is grayscale, we have the option to convert it with a color map
+    if(convertGrayscaleToJetColorMap && inSpec.nchannels == 1)
```

Comment:
in my opinion, we should not perform optional image processing in this plugin, which is an image loader plugin.

Reply:
This image processing block has been removed in 8fd8972